### ISSUE TITLE
@Schema on BigDecimal field causes POJO behavior

### ISSUE THREAD

**Issue by eratio08** [micronaut-projects/micronaut-openapi#564]
### Expected Behavior

In 2.x.x it was

```kotlin
data class Dto(
  @Schema( description="Should become a number in the spec" )
  val shouldBeNumber: BigDecimal
)
```
```yaml
Dto:
 required: 
  - shouldBeNumber
  type: object
  properties:
    shouldBeNumber:
      type: number
      description: Should become a number in the spec
```

### Actual Behaviour

In 3.0.0 the PoJo behavior as describe in https://micronaut-projects.github.io/micronaut-openapi/latest/guide/index.html#schemasAndPojos is applied.

```yaml
Dto:
 required: 
  - shouldBeNumber
  type: object
  properties:
    shouldBeNumber:
       $ref: '#/components/schemas/BigDecimal'
BigDecimal:
      type: object
      properties:
        scale:
          type: integer
          format: int32
        precision:
          type: integer
          format: int32
      description: Should become a number in the spec
```

Leaving out the `@Schema` annotation will lead to the expected behavior.

### Steps To Reproduce

run `gradle clean assemble`

### Environment Information

macOS=11.5.2 (20G95)
java=15.0.2.hs-adpt
kotlin=1.5.30

### Version

3.0.0

**Comment by ilopmar** [comment]
Probably related to #555

**Comment by ilopmar** [comment]
The fix is included in micronaut-openapi 3.0.3. It will be included by default in Micronaut 3.0.2, in the meantime you can force that specific version.
